base/printf.jl

```diff
@@ -13,29 +13,40 @@ const Strings = Union{Val{'s'}, Val{'S'}}
 const Pointer = Val{'p'}
 const HexBases = Union{Val{'x'}, Val{'X'}, Val{'a'}, Val{'A'}}
 
-struct Spec{T} # T => %type => Val{'type'}
+mutable struct Spec{T} # T => %type => Val{'type'}
```

Comment:
It would be better to avoid this if possible. Having the spec be mutable can have performance consequences and is generally just less clear than working with immutable state.

Reply:
Will fix. 

We can avoid having the Spec mutable, the change will then check the dynamic property in each fmt method - not a problem. 

Tests - will add. 

Nota bene - this is my first git pl in a large collaborative project so I'm also learning the in and outs of git.  Things like "adding a cross ref" is still greek to me and I'm not from Greece but will figure out :-)

Reply:
Sorry to be obscure! All I mean is that it's worth adding links (cross references) in your pull request description where relevant so that people can understand the context (not talking about git here). Links like this:

https://github.com/JuliaLang/julia/pull/32859

Reply:
got it :-) 

[#32859](https://github.com/JuliaLang/julia/pull/32859)